src/app/components/routemap/RouteMap.tsx

```diff
@@ -0,0 +1,380 @@
+                continue
+            }
+            arrcyc.push(
+                <RouteLine className={isPrevStop('cycle', i) ? 'bg-chip-red' : 'bg-zinc-200 dark:bg-slate-500'} key={i} ref={d=> d!= null?linecyc.current[i] = d:null}></RouteLine>
+            )
+            arryes.push(
+                <RouteLine className={isPrevStop('yesulin', i) ? 'bg-chip-green' : 'bg-zinc-200 dark:bg-slate-500'} key={i} ref={d=> d!= null?lineyes.current[i] = d:null}></RouteLine>
+            )
+            arrjun.push(
+                <RouteLine className={isPrevStop('jungang', i) ? 'bg-chip-purple' : 'bg-zinc-200 dark:bg-slate-500'} key={i} ref={d=> d!= null?linejun.current[i] = d:null}></RouteLine>
+            )
+        }
+        setDirLine(arrdir)
+        setCycLine(arrcyc)
+        setYesLine(arryes)
+        setJunLine(arrjun)
+    },[isPrevStop])
+
+    const responsiveLines = (refs:React.MutableRefObject<Array<HTMLDivElement>>, lines:React.MutableRefObject<Array<HTMLDivElement>>) => {
+        if(refs.current.length > 0 && lines.current.length > 0){
+            for(let i = 1; i <= lines.current.length; i++){
+                const rectA = {"left":refs.current[i-1]?.offsetLeft, "top":refs.current[i-1]?.offsetTop, "width":refs.current[i-1]?.offsetWidth, "height":refs.current[i-1]?.offsetHeight}
+                const rectB = {"left":refs.current[i]?.offsetLeft, "top":refs.current[i]?.offsetTop, "width":refs.current[i]?.offsetWidth, "height":refs.current[i]?.offsetHeight}
+                const x1 = (rectA.left ?? 0) + (rectA.width ?? 0) / 2
+                const y1 = (rectA.top ?? 0) + (rectA.height ?? 0) / 2
+                const x2 = (rectB.left ?? 0) + (rectB.width ?? 0) / 2
+                const y2 = (rectB.top ?? 0) + (rectB.height ?? 0) / 2
+                
+                const d = Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2))
+                lines.current[i-1].style.width = `${d}px`
+                lines.current[i-1].style.top = `${(refs.current[i-1]?.offsetTop ?? 0)+4}px`
+                const dotchild = refs.current[i-1]?.firstChild as HTMLElement
+                if(dotchild != undefined){
+                    lines.current[i-1].style.left = `${refs.current[i-1]?.offsetLeft+dotchild.offsetLeft+1}px`
+                }
+            }
+        }
+    }
+
+    const updateLines = useCallback(() => {
+        responsiveLines(refdir, linedir)
+        responsiveLines(refcyc, linecyc)
+        responsiveLines(refyes, lineyes)
+        responsiveLines(refjun, linejun)
+    },[])
+
+    const fetchLanguages = useCallback(() => {
+        lang.current.forEach((element:HTMLParagraphElement | null, index:number) => {
+            if(element != undefined){
+                if(index <= 1){
+                    element.innerText = t('yesul')
+                } else {
+                    element.innerText = t('jung')
+                }
+            }
+        })
+    }, [t])
```

Comment:
Setting the innerText directly is a anti pattern in React. Use like all other components did in the other files and let it re-render. By refactoring the codes from the points that I left a review, it will be solved naturally.